Kickstarter-iOS/Views/LiveStreamChatInputView.swift

```diff
@@ -0,0 +1,140 @@
+  func liveStreamChatInputViewRequestedLogin(chatInputView: LiveStreamChatInputView)
+}
+
+internal final class LiveStreamChatInputView: UIView {
+
+  @IBOutlet private weak var moreButton: UIButton!
+  @IBOutlet private weak var rootStackView: UIStackView!
+  @IBOutlet private weak var sendButton: UIButton!
+  @IBOutlet private weak var separatorView: UIView!
+  @IBOutlet private weak var textField: UITextField!
+
+  private weak var delegate: LiveStreamChatInputViewDelegate?
+
+  let viewModel: LiveStreamChatInputViewModelType = LiveStreamChatInputViewModel()
+
+  internal class func fromNib() -> LiveStreamChatInputView {
+    return UINib(nibName: Nib.LiveStreamChatInputView.rawValue, bundle: .framework)
+      .instantiate(withOwner: nil, options: nil)
+      //swiftlint:disable:next force_cast
+      .first as! LiveStreamChatInputView
+  }
+
+  internal func configureWith(delegate: LiveStreamChatInputViewDelegate, chatHidden: Bool) {
+    self.delegate = delegate
+    self.viewModel.inputs.configureWith(chatHidden: chatHidden)
+  }
+
+  internal override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> UIView.lens.backgroundColor .~ UIColor.hex(0x353535)
+
+    _ = self.separatorView
+      |> UIView.lens.backgroundColor .~ UIColor.white.withAlphaComponent(0.2)
+
+    _ = self.rootStackView
+      |> UIStackView.lens.layoutMarginsRelativeArrangement .~ true
+      |> UIStackView.lens.layoutMargins .~ .init(leftRight: Styles.grid(2))
+      |> UIStackView.lens.spacing .~ Styles.grid(2)
+
+    _ = self.textField
+      |> UITextField.lens.backgroundColor .~ UIColor.hex(0x353535)
+      |> UITextField.lens.tintColor .~ .white
+      |> UITextField.lens.textColor .~ .white
+      |> UITextField.lens.font .~ .ksr_body(size: 14)
+      |> UITextField.lens.borderStyle .~ .none
+      |> UITextField.lens.returnKeyType .~ .done
+
+    _ = self.sendButton
+      |> UIButton.lens.tintColor .~ .white
+      |> UIButton.lens.title(forState: .normal) .~ localizedString(key: "Send", defaultValue: "Send")
+
+    _ = self.moreButton
+      |> UIButton.lens.tintColor .~ .white
+  }
+
+  internal override func bindViewModel() {
+    super.bindViewModel()
+
+    self.moreButton.addTarget(self, action: #selector(LiveStreamChatInputView.more), for: .touchUpInside)
+    self.sendButton.addTarget(self, action: #selector(LiveStreamChatInputView.send), for: .touchUpInside)
+
+    self.moreButton.rac.hidden = self.viewModel.outputs.moreButtonHidden
+    self.sendButton.rac.hidden = self.viewModel.outputs.sendButtonHidden
+
+    self.viewModel.outputs.notifyDelegateMoreButtonTapped.observeValues { [weak self] in
+      self.flatMap { $0.delegate?.liveStreamChatInputViewDidTapMoreButton(chatInputView: $0) }
+    }
+
+    self.viewModel.outputs.notifyDelegateMessageSent
+      .observeForUI()
+      .on(value: { [weak self] _ in
+        self?.textField.text = nil
+        self?.textField.resignFirstResponder()
+      })
```

Comment:
let's not inject side effects like this. should this instead be a separate output?